plugin/core/sessions.py

```diff
@@ -1771,12 +1771,16 @@ def apply_workspace_edit_async(self, edit: WorkspaceEdit) -> Promise[None]:
         return self.apply_parsed_workspace_edits(parse_workspace_edit(edit))
 
     def apply_parsed_workspace_edits(self, changes: WorkspaceChanges) -> Promise[None]:
+        active_sheet = self.window.active_sheet()
+        selected_sheets = self.window.selected_sheets()
         promises = []  # type: List[Promise[None]]
         for uri, (edits, view_version) in changes.items():
             promises.append(
                 self.open_uri_async(uri).then(functools.partial(self._apply_text_edits, edits, view_version, uri))
             )
-        return Promise.all(promises).then(lambda _: None)
+        return Promise.all(promises).then(
+            lambda _: self._check_select_sheets(selected_sheets)).then(
+            lambda _: self._check_focus_sheet(active_sheet))
```

Comment:
Would prefer this style:

```suggestion
        return Promise.all(promises)
            .then(lambda _: self._check_select_sheets(selected_sheets))
            .then(lambda _: self._check_focus_sheet(active_sheet))
```

Reply:
I initially wanted to write it like that, but this is not valid Python code.

It would need to have explicit line continuations `\` on the two lines, but then it looks even more ugly as it is now (IMO).

Reply:
We do use that syntax with `\` and I think it's more readable since then you see each chain start on the new line.
But I won't fight over it.

Reply:
There are a few places in the code where the style from this PR is used, but the explicit line continuation is more common. I'll unify the styles used.